### PR TITLE
Added a debug command and some logging to the RC anchors reloading

### PR DESCRIPTION
**Pull request by repo-alt**

### CODE REVIEW

**Review by mitchej123 — APPROVED** [review]

File: src/main/java/com/mitchej123/hodgepodge/core/commands/DebugCommand.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            Stream.of("toggle", "anchor")
+                .filter(s -> test.isEmpty() || s.startsWith(test))
+                .forEach(l::add);
+        } else if (test.equals("toggle")) {
+            String test1 = ss[1].trim();
+            if (test1.isEmpty() || "anchordebug".startsWith(test1))
+                l.add("anchordebug");
+        } else if (test.equals("anchor")) {
+            String test1 = ss[1].trim();
+            if (test1.isEmpty() || "list".startsWith(test1))
```
Comment (Glease) [inline src/main/java/com/mitchej123/hodgepodge/core/commands/DebugCommand.java:43]: Missing a check for situation, where admin types `/hp anchor list ` then press tab, expecting to see a list of usernames as completion.

FYI use `getListOfStringsMatchingLastWord(as, MinecraftServer.getServer().getAllUsernames())` to quickly get a list of possible completions of username.

File: src/main/java/com/mitchej123/hodgepodge/core/commands/DebugCommand.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    @Override
+    public String getCommandUsage(ICommandSender sender) {
+        return "Usage: hp <subcommand>. Valid subcommands are: toggle, anchor.";
+    }
+    private void printHelp(ICommandSender sender) {
+        sender.addChatMessage(new ChatComponentText("Usage: hp <toggle|anchor>"));
+        sender.addChatMessage(new ChatComponentText("\"toggle anchordebug\" - toggles RC anchor debugging"));
+        sender.addChatMessage(new ChatComponentText("\"anchor list <player>\" - list RC anchors placed by the player (empty for current player)"));
```
Comment (Glease) [inline src/main/java/com/mitchej123/hodgepodge/core/commands/DebugCommand.java:27]: Could have been `"\"anchor list [player]\" - list RC anchors placed by the player (empty for current player)"`

File: src/main/java/com/mitchej123/hodgepodge/core/commands/DebugCommand.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                AnchorAlarm.AnchorDebug = !AnchorAlarm.AnchorDebug;
+                sender.addChatMessage(new ChatComponentText("Anchor debugging: " + AnchorAlarm.AnchorDebug));
+                break;
+            case "anchor":
+                if (strings.length < 2 || !strings[1].equals("list")) {
+                    printHelp(sender);
+                    return;
+                }
+                String playerName = strings.length > 2 ? strings[2] : sender.getCommandSenderName();
+                if (!AnchorAlarm.listSavedAnchors(playerName, sender.getEntityWorld()))
```
Comment (Glease) [inline src/main/java/com/mitchej123/hodgepodge/core/commands/DebugCommand.java:69]: I usually use `net.minecraft.server.management.ServerConfigurationManager#playerEntityList` to find out all players. Not sure if these are equivalent, just saying.

**Review by Glease — COMMENTED** [review]
Minor problem in tab completion, but otherwise looks good.

**Comment by Dream-Master** [comment]
oh i was to fast ?

**Comment by Glease** [comment]
nothing game breaking. could be in another PR.

File: src/main/java/com/mitchej123/hodgepodge/core/commands/DebugCommand.java (hunk 1 above)
Comment (repo-alt) [reply src/main/java/com/mitchej123/hodgepodge/core/commands/DebugCommand.java:43]: ok

File: src/main/java/com/mitchej123/hodgepodge/core/commands/DebugCommand.java (hunk 3 above)
Comment (repo-alt) [reply src/main/java/com/mitchej123/hodgepodge/core/commands/DebugCommand.java:69]: getAllUsernames literally calls serverConfigurationManager

**Pull request merged**